### PR TITLE
XRDDEV-1172 WSDL request size limit bug

### PR DESCRIPTION
**Pull request by carohauta**
The bug could not be reproduced but there was a missing feature that was implemented in this PR: 
10MB size limit for WSDLs

Check the task description for more comments:
JIRA: https://jira.niis.org/browse/XRDDEV-1172

### CODE REVIEW

File: src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/wsdl/WsdlParser.java [hunk 1]
```diff
@@ -282,22 +288,35 @@ public ServiceInfo(String name, String title, String url,
...
+                        byteArrayOutputStream.write(buf, 0, n);
+                    }
                 }
+                byte[] response = byteArrayOutputStream.toByteArray();
                 log.trace("Received WSDL response: {}", new String(response));
 
                 return new InputSource(new ByteArrayInputStream(response));
+            } catch (WsdlParseException e) {
+                throw e;
             } catch (Throwable t) {
```
Comment (jhyoty) [inline src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/wsdl/WsdlParser.java:320]: This catches also java.lang.Errors, which should not be normally done. Should be narrowed to the actual possible checked exception types or at least "Exception".

File: src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/wsdl/WsdlParser.java [hunk 2]
```diff
@@ -282,22 +288,35 @@ public ServiceInfo(String name, String title, String url,
...
+                        count += n;
+                        if (count > MAX_DESCRIPTION_SIZE) {
+                            throw new WsdlParseException(
+                                    "Error reading WSDL: Size exceeds " + MAX_DESCRIPTION_SIZE + " bytes.");
+                        }
+                        byteArrayOutputStream.write(buf, 0, n);
+                    }
                 }
+                byte[] response = byteArrayOutputStream.toByteArray();
                 log.trace("Received WSDL response: {}", new String(response));
```
[the remark this answers is not in the document]
Comment (carohauta) [reply src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/wsdl/WsdlParser.java]: Added the suggested if clause

File: src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/wsdl/WsdlParser.java [hunk 3]
```diff
@@ -282,22 +288,35 @@ public ServiceInfo(String name, String title, String url,
...
-                byte[] response;
                 try (InputStream in = conn.getInputStream()) {
-                    response = IOUtils.toByteArray(in);
+                    long count = 0;
+                    int n;
+                    byte[] buf = new byte[BUF_SIZE];
+                    while ((n = in.read(buf)) != -1) {
+                        count += n;
+                        if (count > MAX_DESCRIPTION_SIZE) {
+                            throw new WsdlParseException(
```
[the remark this answers is not in the document]
Comment (carohauta) [reply src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/wsdl/WsdlParser.java]: Changed to `UncheckedIOException`

File: src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/wsdl/WsdlParser.java (hunk 1 above)
Comment (carohauta) [reply src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/wsdl/WsdlParser.java]: Updated to catch `Exception`s

**Pull request merged**